### PR TITLE
Added automatic push of docs to scitools-docs/iris.

### PR DESCRIPTION
**Pull request by pelson**
**Notice**: this is a SciTools/iris branch, and should be deleted once merged.

I've added a final step to our doctest build matrix to push the built docs to https://github.com/SciTools-docs/iris. The setup currently is:

 * push a folder named by the branch being built (e.g master) to the gh-pages branch
 * Python 3 is used as doctr only supports 3.5+. This means we aren't pushing any of the GRIB stuff currently (not Py3 compatible).
 * I've used a separate organisation for several reasons:
    * it isolates the impact of a token being leaked accidentally (nobody can modify SciTools/iris itself
    * it prevents the automatic building of docs pushing a repository artificially high up the list of repositories in github.com/SciTools
    * We don't need to worry about the size of the repository - it is Iris' docs only, and nobody normally will be cloning it (though there is a question in my mind about performance that says that doctr would need to *pull* the repository)
    * It is easy to rinse and repeat for other project docs (e.g. cartopy)
    * The downsides:
        * Another organisation, with an entirely different set of teams etc. (still, most people don't actually need write access to it)
        * Falls under a different domain scitools-docs.github.io. We *could* make this a subdomain of scitools.org.uk (e.g. docs.scitools.org.uk)
        * It isn't readthedocs (we are likely to have hit time/size limits with that, and there is an advertising program that, whilst reasonable, doesn't sit too comfortably with me)

The results of this can be seen in https://github.com/SciTools-docs/iris and https://scitools-docs.github.io/iris/travis-docs/. (this is also the reason for the branch being on SciTools)

### CODE REVIEW

**Comment by pelson** [comment]
Also, we will want to:

- [ ] manually delete the travis-docs folder in the scitools-docs organisation
- [ ] ensure that more people than just me have access to scitools-docs

**Comment by pelson** [comment]
Looks like there is a genuine problem with master, not with this change.

**Comment by cpelley** [comment]
> It isn't readthedocs
Was your original motivations around readthedocs about its abilities around versioning?  Something else?

**Comment by bjlittle** [comment]
@pelson I'm working on fixing master now ... mostly changes from `numpy` bumping from `1.12.1` to `1.13.0`, and there may be fall-out from `netcdf4` bumping from `1.2.8` to `1.2.9` ...

**Comment by pelson** [comment]
> Was your original motivations around readthedocs about its abilities around versioning? Something else?

I'm not 100% I've understood, but here are my (very opinionated) thoughts on RTD:

 * It adds a layer of complexity to build Iris' docs on RTD's infra (not RTD's fault, we have an unreasonably/unnecessarily complex doc build process)
 * we already build the docs with every commit, we don't need to do that work again
 * advertising is not something I'm especially keen to add to our docs (though I have no problem with RTD itself having done so - it makes a lot of sense to help them cover the substantial costs of operation)
 * doctr made it sustainable to maintain code to push from a repository (I've written my fair share of this kind of code, but didn't have the space to generalise as has been done in doctr)

To be clear, I'm only proposing this change as a means of seeing *dev* docs. We can talk about tagged builds, but that falls out of scope for this initial PR.

**Comment by cpelley** [comment]
Thanks @pelson, cool!

**Comment by bjlittle** [comment]
@pelson Awesome! :+1: 

Rebase when #2616 is merged, and let's get this merged :smile:

**Comment by pelson** [comment]
Rebased.

**Comment by pp-mo** [comment]
And we now have ... https://scitools-docs.github.io/iris/master/ 
Which is simply brilliant ! :champagne: :1st_place_medal:

**Comment by pelson** [comment]
Thanks for reviewing and merging.

File: .travis.yml [hunk 1]
```diff
@@ -117,15 +122,30 @@ script:
   # spuriously. The echo-backtick workaround gets around this error,
   # which should be investigated further in the future.
   - if [[ $TEST_TARGET == 'doctest' ]]; then
-      cd $(pwd)/docs/iris;
+      cd docs/iris;
       echo `make clean html`;
       make doctest;
+      cd ../../;
```
Comment (pp-mo) [inline .travis.yml]: Prevents any test failure being recognised by Travis !

File: .travis.yml [hunk 2]
```diff
@@ -117,15 +122,30 @@ script:
...
+        ${TRAVIS_BRANCH:-${TRAVIS_TAG}};
+    fi
+
   # An extra call to check "whatsnew" contributions are valid, because the
   # Iris test for it needs a *developer* install to be able to find the docs.
   - if [[ $TEST_TARGET == 'doctest' ]]; then
-      cd ./src/whatsnew;
+      cd docs/iris/src/whatsnew;
       python aggregate_directory.py --checkonly;
+      cd ../../../../;
```
Comment (pp-mo) [inline .travis.yml]: Prevents any test failure being recognised by Travis !

**Pull request merged**